File: backend/app/schemas/reputation.py
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class ReputationProfileResponse(BaseModel):
    user_id: UUID

    trust_score: Decimal
    trust_level: str

    completed_transactions: int
    completed_as_buyer: int
    completed_as_seller: int

    cancelled_transactions: int

    review_count: int
    average_rating: Decimal | None

    phone_verified: bool
    identity_verified: bool
    business_verified: bool

    model_config = ConfigDict(
        from_attributes=True
    )

File: frontend/src/types/reputation.ts
export type ReputationProfile = {
  user_id: string;
  trust_score: string;
  trust_level: string;
  completed_transactions: number;
  completed_as_buyer: number;
  completed_as_seller: number;
  cancelled_transactions: number;
  review_count: number;
  average_rating: string | null;
  phone_verified: boolean;
  identity_verified: boolean;
  business_verified: boolean;
};

export type UserReview = {
  id: string;
  transaction_id: string;
  reviewer_id: string;
  reviewed_user_id: string;
  rating: number;
  comment: string | null;
  status: string;
  created_at: string;
};
